packages/mongo-nav/src/mongo-select/ProjectSelect.tsx

```diff
@@ -198,6 +199,12 @@ function ProjectSelect({
   const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const value = e.target.value;
     setValue(value);
+
+    if (open === true && wasOpen !== open) {
+      // Opt out of type-checking as we just want to trigger the onElementClick handler
+      onElementClick(NavElement.ProjectNavProjectSelectSearch)(e as any);
```

Comment:
I feel like I'm missing something, but what's the reason for treating this onChange event as an onClick from the POV of the consumer?

Reply:
Simply? Ease.

I could expose another handler for notifying when this event occurs but that's going to involve a lot more conversation.

Reply:
My primary worry here is that we may want to deprecate this event firing in the future since it's inconsistent with the name and other uses of the prop. If that's acceptable to ya, I'm fine with it.

Reply:
You mean in general? And replace it with what? 

I mean I'm fine for this to fit here awkwardly or I can add a new prop..but that seems so much more painful.

Or I can just remove this code for now and deal with it in a follow-up ticket.

Reply:
To clarify, I don't feel too strongly about this, just that I could see this being a point of 
 confusion in the future because the name of the prop doesn't reflect the usage.

We can "awkwardly" add this here for now, though in the longer term I expect us to want to do one of the following:
1. We create an `onElementChange` prop that a consumer can hook into.
2. We rename and restructure the prop to something like `onNavEvent(navElement, eventType)`.

Reply:
I'm fine to change that in the future. When we upgrade mongo nav we'll just change the handler we use. 

This is just for analytics so I don't care terribly.